libdino/src/entity/account.vala

```diff
@@ -57,6 +69,37 @@ public class Account : Object {
         notify.connect(on_update);
     }
 
+    public async string? get_password() {
+#if WITH_SECRET
+        if (password == "") {
+            // We have to look it up in the secret service
+            return yield secret_manager.get_password();
+        }
+
+        // We have a clear-text password.  Try to put it into the keyring.
+        var cleartext_password = password;
+        if (yield secret_manager.set_password(password)) {
+            // Success.  Clear the clear-text PW from this entity.
+            this.password = "";
+        }
+        // Either way, return the password.
+        return cleartext_password;
+#else
+        return this.password;
+#endif
+    }
+
+    public async void set_password(string password) {
+#if WITH_SECRET
+        if (yield secret_manager.set_password(password)) {
+            // Success.  Don't keep anything locally.
+            this.password = "";
+            return;
+        }
+#endif
+        this.password = password;
```

Comment:
I’m uneasy with silently falling back to plain text storage when libsecret failed, but I don’t have a good solution to suggest.

Reply:
In the future, maybe this function could inform its caller how the password was stored, and then the account UI could inform the user. But my UI experience is very limited (ahem ;), so I focused on the core functionality.